src/clj/rems/db/attachments.clj

```diff
@@ -25,6 +29,42 @@
      :attachment/data data
      :attachment/type type}))
 
+(defn scan-for-malware
+  "Feeds byte-array to STDIN and runs executable at malware-scanner-path returns true if malware executable returns a non-zero status-code, false otherwise, logs STERR of executable"
+  [malware-scanner-path byte-array]
+  (let [scan-output (sh "sh" "-c" malware-scanner-path :in byte-array)]
+    {:detected (not= (:exit scan-output) 0)
+     :log (:err scan-output)}))
+
+(deftest test-passing-scan-without-output
+  (let [scan (scan-for-malware "test-data/malware-scanner-executables/pass-without-output.sh" "")]
+    (is (and (not (:detected scan)) (nil? (:log scan))))))
+
+(deftest test-failing-scan-without-output
+  (let [scan (scan-for-malware "test-data/malware-scanner-executables/fail-without-output.sh" "")]
+    (is (and (:detected scan) (nil? (:log scan))))))
+
+(deftest test-passing-scan-with-output
+  (let [scan (scan-for-malware "test-data/malware-scanner-executables/pass-with-output.sh" "")]
+    (is (and (not (:detected scan)) (= (:log scan) "passed")))))
+
+(deftest test-failing-scan-with-output
+  (let [scan (scan-for-malware "test-data/malware-scanner-executables/fail-with-output.sh" "")]
+    (is (and (:detected scan) (= (:log scan) "failed")))))
+
+(deftest test-scanner-receives-input
+  (let [input "miauw"
+        scan (scan-for-malware "test-data/malware-scanner-executables/cat-to-stderr.sh" input)]
+    (is (and (not (:detected scan)) (= (:log scan) input)))))
+
+(defn check-for-malware-if-enabled [byte-array]
+  (when-let [malware-scanner-path (str (:malware-scanner-path env))]
```

Comment:
We can leave the `str` from here, the path should be a string or `nil`.

Reply:
fixed in [b0e72a5](https://github.com/CSCfi/rems/pull/3202/commits/b0e72a5d50b69d4b3512eb87d4d4e334cc00b6f4)